docs/Displaying a User Interface/Main Screen Templates/index.md

```diff
@@ -1,76 +1,83 @@
+
 
 @![iOS]
 ##### Objective-C
 ```objc
-SDLSetDisplayLayout* display = [[SDLSetDisplayLayout alloc] initWithPredefinedLayout:SDLPredefinedLayoutGraphicWithText];
-[self.sdlManager sendRequest:display withResponseHandler:^(SDLRPCRequest *request, SDLRPCResponse *response, NSError *error) {
-     if (!response.success.boolValue) { 
-        // Print out the error if there is one and return early
-        return;
-     }
-     // The template has been set successfully
-}];
+//TODO
 ```
 
 ##### Swift
 ```swift
-let display = SDLSetDisplayLayout(predefinedLayout: .graphicWithText)
-sdlManager.send(request: display) { (request, response, error) in
-    guard response?.success.boolValue == true else { return }
-        // The template has been set successfully
-}
+//TODO
 ```
 !@
 
 @![android, javaSE, javaEE]
 ```java
-SetDisplayLayout setDisplayLayoutRequest = new SetDisplayLayout()
-    .setDisplayLayout(PredefinedLayout.GRAPHIC_WITH_TEXT.toString());
-
-setDisplayLayoutRequest.setOnRPCResponseListener(new OnRPCResponseListener() {
+TemplateConfiguration templateConfiguration = new TemplateConfiguration().setTemplate(PredefinedLayout.GRAPHIC_WITH_TEXT.toString());
+sdlManager.getScreenManager().changeLayout(templateConfiguration, new CompletionListener() {
     @Override
-    public void onResponse(int correlationId, RPCResponse response) {
-        if (((SetDisplayLayoutResponse) response).getSuccess()) {
-            Log.i("SdlService", "Display layout set successfully.");
+    public void onComplete(boolean success) {
+        if (success) {
+            DebugTool.logInfo(TAG, "Layout set successfully");
         } else {
-            Log.i("SdlService", "Display layout request rejected.");
+            DebugTool.logInfo(TAG, "Layout not set successfully");
         }
     }
 });
-
-sdlManager.sendRPC(setDisplayLayoutRequest);
 ```
 !@
 
 @![javascript]
 ```js
-// sdl_javascript_suite v1.1+
-const setDisplayLayoutRequest = new SDL.rpc.messages.SetDisplayLayout();
-setDisplayLayoutRequest.setDisplayLayout(SDL.rpc.enums.PredefinedLayout.GRAPHIC_WITH_TEXT);
-const response = await sdlManager.sendRpcResolve(alert);
-if (response.getSuccess()) {
-    console.log('Display layout set successfully.');
-} else {
-    console.log('Display layout request rejected.');
-}
-// thrown exceptions should be caught by a parent function via .catch()
-
-// Pre sdl_javascript_suite v1.1
-const setDisplayLayoutRequest = new SDL.rpc.messages.SetDisplayLayout();
-setDisplayLayoutRequest.setDisplayLayout(SDL.rpc.enums.PredefinedLayout.GRAPHIC_WITH_TEXT);
-const response = await sdlManager.sendRpc(setDisplayLayoutRequest).catch(function (error) {
-    // Handle Error
+//TODO
+```
+!@
+
+Template changes can also be batched with text and graphics updates.
+
+@![iOS]
+##### Objective-C
+```objc
+//TODO
```

Comment:
```suggestion
[self.sdlManager.screenManager beginUpdates];
self.sdlManager.screenManager.textField1 = "Line of Text";
[self.sdlManager.screenManager changeLayout:[[SDLTemplateConfiguration alloc] initWithTemplate:SDLPredefinedLayoutGraphicWithText] withCompletionHandler:^(NSError * _Nullable error) {
    // This listener will be ignored, and will use the handler sent in commit.
}];
self.sdlManager.screenManager.primaryGraphic = <#SDLArtwork#>;
[self.sdlManager.screenManager endUpdatesWithCompletionHandler:^(NSError * _Nullable error) {
    if (error != nil) {
        // Print out the error if there is one and return early
        return
    }
    // The data and template has been set successfully
}];
```

Reply:
For:
```
// This listener will be ignored, and will use the handler sent in commit.

```
Instead of commit should it be endUpdatesWithCompletionHandler